pkg/alertmanager/storage.go

```diff
@@ -0,0 +1,40 @@
+package alertmanager
+
+import (
+	"flag"
+	"fmt"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alerts"
+	"github.com/cortexproject/cortex/pkg/alertmanager/local"
+	"github.com/cortexproject/cortex/pkg/configs/client"
+)
+
+// AlertStoreConfig configures the alertmanager backend
+type AlertStoreConfig struct {
+	Type     string `yaml:"type"`
+	ConfigDB client.Config
+	File     local.FileAlertStoreConfig `yaml:"file_config"`
+}
+
+// RegisterFlags registers flags.
+func (cfg *AlertStoreConfig) RegisterFlags(f *flag.FlagSet) {
+	cfg.File.RegisterFlags(f)
+	cfg.ConfigDB.RegisterFlagsWithPrefix("alertmanager.", f)
+	f.StringVar(&cfg.Type, "alertmanager.storage.type", "configdb", "Method to use for backend rule storage (configdb, file)")
```

Comment:
I think everything related to backend config storage should have a `alertmanager.storage.`. However, I wanted to avoid making a breaking change in this PR and updating the prefix for `alertmanager.configs` to `alertmanager.storage.configs` would force users to change their flags. I can add a TODO to make that change in the future and go throough a change/deprecation process in a future PR?